From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Subject: [PATCH] Add missing sockaddr cast
Date: Wed,  3 Jul 2024 08:17:02 +0200

Build log:
	port-linux.c: In function 'ssh_systemd_notify':
	port-linux.c:369:25: error: passing argument 2 of 'connect' from incompatible pointer type [-Wincompatible-pointer-types]
	  369 |         if (connect(fd, &addr, sizeof(addr)) != 0) {
	      |                         ^~~~~
	      |                         |
	      |                         struct sockaddr_un *
	In file included from ../includes.h:26,
	                 from port-linux.c:22:
	/usr/include/sys/socket.h:386:19: note: expected 'const struct sockaddr *' but argument is of type 'struct sockaddr_un *'
	  386 | int connect (int, const struct sockaddr *, socklen_t);
	      |                   ^~~~~~~~~~~~~~~~~~~~~~~
	make[1]: *** [Makefile:106: port-linux.o] Error 1

Upstream-Status: Pending
Origin: Source Mage
Signed-off-by: Ismael Luceno <ismael@sourcemage.org>
---
 openbsd-compat/port-linux.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/openbsd-compat/port-linux.c b/openbsd-compat/port-linux.c
index 566e3378c1c4..86300517829b
--- a/openbsd-compat/port-linux.c
+++ b/openbsd-compat/port-linux.c
@@ -366,7 +366,7 @@
 		error_f("socket \"%s\": %s", path, strerror(errno));
 		goto out;
 	}
-	if (connect(fd, &addr, sizeof(addr)) != 0) {
+	if (connect(fd, (struct sockaddr *)&addr, sizeof(addr)) != 0) {
 		error_f("socket \"%s\" connect: %s", path, strerror(errno));
 		goto out;
 	}
-- 
Fixed up by sm-checkpatch 0.1

